### PR TITLE
Condense container library

### PR DESCRIPTION
**Pull request by jpreese**
After upgrading to Konstraint `v0.6.0` which contains support for recursively adding imports, I noticed that my policies were now updated to include the `pods` library. However, they all worked previously without needing to import it before.

Giving the libraries a closer look, noticed that a large portion of the pods/containers pieces can be shortened.

Any thoughts on this / something similar? This would remove the need for `containers.rego` and use `pods.rego`, now that its incredibly small.

I also like how the library is called now, instead of `containers.containers` and `pods.pods` -- we just have `pods.containers[_]`

### CODE REVIEW

File: examples/lib/pods.rego [hunk 1]
```diff
@@ -2,32 +2,21 @@ package lib.pods
 
 import data.lib.core
 
-pods[pod] {
-    lower(core.kind) == "statefulset"
-    pod = core.resource.spec.template
+pod = core.resource.spec.template {
+    pod_templates := ["daemonset","deployment","job","statefulset"]
```
Comment (jalseth) [inline examples/lib/pods.rego:6]: We probably should have done this earlier, but we should add `replicaset` as well. Even though `Depoyment`s are standard now, we don't want folks to bypass our policies -- especially our security policies -- by using a different (edit: but still valid) resource type. 

We're also missing support for CronJob resources but those use a different format than `resource.spec.template`.

Edit again: ReplicationController, added in 1.18, should also be included.

**Comment by jpreese** [comment]
Good catch on ReplicationController-- didn't even realize that's a thing now. I added that and CronJob.

File: examples/lib/pods.rego (hunk 1 above)
Comment (jpreese) [reply examples/lib/pods.rego]: Done!

File: examples/lib/pods.rego [hunk 2]
```diff
@@ -2,32 +2,25 @@ package lib.pods
 
 import data.lib.core
 
-pods[pod] {
-    lower(core.kind) == "statefulset"
-    pod = core.resource.spec.template
+pod = core.resource.spec.template {
+    pod_templates := ["daemonset","deployment","job","replicaset","replicationcontroller","statefulset"]
```
Comment (garethahealy) [inline examples/lib/pods.rego:6]: would you guys be interested in supporting the OpenShift `DeploymentConfig` kind as well?

Comment (jpreese) [reply examples/lib/pods.rego:6]: I'm not entirely against the idea. I would have concerns about considering _every_ edge case and having the library spiral out of control--but just a single case isn't going to be that big of a problem. In a perfect scenario, we'd have an `openshift.rego` library that implemented `core` and added additional pod definitions (and other openshift specific definitions).

@jalseth , any thoughts on this one?

Comment (jalseth) [reply examples/lib/pods.rego:6]: I'd prefer if we added an `lib.openshift` that was a shim for `lib.core` with OpenShift specific items added. However, at this point, adding a shim just to match on one more Kind is probably excessive. We should revisit this when any other OpenShift items need to be added to the libraries.

**Pull request merged**